plutus-tx/src/PlutusTx/Builtins.hs

```diff
@@ -382,6 +383,10 @@ encodeUtf8 = BI.encodeUtf8
 matchList :: forall a r . BI.BuiltinList a -> r -> (a -> BI.BuiltinList a -> r) -> r
 matchList l nilCase consCase = BI.chooseList l (const nilCase) (\_ -> consCase (BI.head l) (BI.tail l)) ()
 
+{-# INLINE headMaybe #-}
+headMaybe :: BI.BuiltinList a -> Maybe a
+headMaybe l = matchList l Nothing (\h _ -> Just h)
+
```

Comment:
I think the general consensus in the community is naming this `safeHead` ?

Reply:
I've seen both, I think?